tests/base.py

```diff
@@ -39,47 +41,84 @@ def setUp(self):
         )
 
 
-class ArticleTestBase(TestCase):
+class RequireBasicData(RequireSuperuserMixin):
     """
-    Sets up basic data for testing with an article and some revisions
+    Mixin that creates common data required for all tests.
     """
+    pass
 
-    def setUp(self):
 
-        super(ArticleTestBase, self).setUp()
+class TestBase(RequireBasicData, TestCase):
+    pass
+
 
+class RequireRootArticleMixin(object):
+
+    def setUp(self):
+        super(RequireRootArticleMixin, self).setUp()
         self.root = URLPath.create_root()
-        self.child1 = URLPath.create_article(self.root, 'test-slug', title="Test 1")
+        self.root_article = URLPath.root().article
+        rev = self.root_article.current_revision
+        rev.title = "Root Article"
+        rev.content = "root article content"
+        rev.save()
 
 
-class WebTestBase(TestBase):
+class ArticleTestBase(RequireRootArticleMixin, TestBase):
+    """
+    Sets up basic data for testing with an article and some revisions
+    """
+    pass
 
-    def setUp(self):
-        """Login as the superuser created because we shall access restricted
-        views"""
 
-        super(WebTestBase, self).setUp()
+# This is needed only for tests that use the Django test client,
+# which are being phased out.
+
+class DjangoClientTestBase(TestBase):
+    def setUp(self):
+        super(DjangoClientTestBase, self).setUp()
 
         self.c = c = Client()
         c.login(username=SUPERUSER1_USERNAME, password=SUPERUSER1_PASSWORD)
 
 
-class ArticleWebTestBase(WebTestBase):
-
-    """Base class for web client tests, that sets up initial root article."""
 
+class WebTestCommonMixin(RequireBasicData, django_functest.ShortcutLoginMixin):
+    """
+    Common setup required for WebTest and Selenium tests
+    """
     def setUp(self):
+        super(WebTestCommonMixin, self).setUp()
 
-        super(ArticleWebTestBase, self).setUp()
+        self.shortcut_login(username=SUPERUSER1_USERNAME,
+                            password=SUPERUSER1_PASSWORD)
 
-        response = self.c.post(
-            reverse('wiki:root_create'),
-            {'content': 'root article content', 'title': 'Root Article'},
-            follow=True
-        )
 
-        self.assertEqual(response.status_code, 200)  # sanity check
-        self.root_article = URLPath.root().article
+class WebTestBase(WebTestCommonMixin, django_functest.FuncWebTestMixin, TestCase):
+    pass
+
+
+INCLUDE_SELENIUM_TESTS = os.environ.get('INCLUDE_SELENIUM_TESTS', '0') == '1'
+
+
+@unittest.skipIf(not INCLUDE_SELENIUM_TESTS, "Skipping Selenium tests")
+class SeleniumBase(WebTestCommonMixin, django_functest.FuncSeleniumMixin, StaticLiveServerTestCase):
+    driver_name = "Chrome"
```

Comment:
Guessing at some point this will be made configurable, but there's no need at present.